src/prif.f90

```diff
@@ -49,55 +57,8 @@ module prif
     type(handle_data), pointer :: coarrays
   end type
 
-  interface prif_atomic_cas
-     module procedure prif_atomic_cas_int
-     module procedure prif_atomic_cas_logical
-  end interface
-
-  interface prif_atomic_define
-     module procedure prif_atomic_define_int
-     module procedure prif_atomic_define_logical
-  end interface
-
-  interface prif_atomic_ref
-     module procedure prif_atomic_ref_int
-     module procedure prif_atomic_ref_logical
-  end interface
-
-  interface prif_lcobound
-     module procedure prif_lcobound_with_dim
-     module procedure prif_lcobound_no_dim
-  end interface
-
-  interface prif_ucobound
-     module procedure prif_ucobound_with_dim
-     module procedure prif_ucobound_no_dim
-  end interface
-
-  interface prif_this_image
-
-    module subroutine prif_this_image_no_coarray(team, image_index)
-      implicit none
-      type(prif_team_type), intent(in), optional :: team
-      integer(c_int), intent(out) :: image_index
-    end subroutine
-
-    module subroutine prif_this_image_with_coarray(coarray_handle, team, cosubscripts)
-      implicit none
-      type(prif_coarray_handle), intent(in) :: coarray_handle
-      type(prif_team_type), intent(in), optional :: team
-      integer(c_intmax_t), intent(out) :: cosubscripts(:)
-    end subroutine
-
-    module subroutine prif_this_image_with_dim(coarray_handle, dim, team, cosubscript)
-      implicit none
-      type(prif_coarray_handle), intent(in) :: coarray_handle
-      integer(c_int), intent(in) :: dim
-      type(prif_team_type), intent(in), optional :: team
-      integer(c_intmax_t), intent(out) :: cosubscript
-    end subroutine
-
-  end interface
+  integer(c_int), parameter, public :: PRIF_ATOMIC_INT_KIND = selected_int_kind(18)
+  integer(c_int), parameter, public :: PRIF_ATOMIC_LOGICAL_KIND = PRIF_ATOMIC_INT_KIND
```

Comment:
The use of an integer kind as a logical kind is definitely not portable, but I kind of think we should use the C kinds here (corresponding to what GASNet uses for atomic operations).

Reply:
@bonachea , says use `selected_logical_kind(32)`, and I agree.

Reply:
```suggestion
  integer(c_int), parameter, public :: PRIF_ATOMIC_INT_KIND = selected_int_kind(18)
  integer(c_int), parameter, public :: PRIF_ATOMIC_LOGICAL_KIND = selected_logical_kind(32)
```

Reply:
I don't have any idea what their rationale would have been for specifying it that way, but I doubt I would have agreed with it.

Reply:
IIUC @everythingfunctional 's motivation for this change is because int kind parameters are not guaranteed to be meaningful logical kind parameters. So the old code was very non-portable.

However I think it's worth noting that even this new implementation assumes the compiler HAS a logical type of at least 32 bits. Hopefully flang provides that?

IIUC a Fortran compiler could(?) be compliant providing only an 8-bit logical kind, and such a compiler would return `-1` above, rendering Caffeine erroneous. If we discover there is such a compiler of interest out there, we'd need considerably more effort to support logical atomics, and we'd probably need some configure-time knobs to activate that support.

Reply:
> IIUC @everythingfunctional 's motivation for this change is because int kind parameters are not guaranteed to be meaningful logical kind parameters. So the old code was very non-portable.

Correct.

> However I think it's worth noting that even this new implementation assumes the compiler HAS a logical type of at least 32 bits.

> IIUC a Fortran compiler could(?) be compliant providing only an 8-bit logical kind, and such a compiler would return `-1` above, rendering Caffeine erroneous.

This is true, but seems unlikely to occur in practice, and we can probably cross that bridge when (if) we come to it.

Reply:
Just to summarize conversations we had elsewhere related to this: gfortran doesn't support `selected_logical_kind`. As such, this code suggestion has been reverted and then a comment documenting this has been added to the code.